pipeline/execout/cachev1/exec.go

```diff
@@ -0,0 +1,56 @@
+package cachev1
+
+import (
+	"fmt"
+	"github.com/streamingfast/bstream"
+	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
+	"github.com/streamingfast/substreams/pipeline/execout"
+	"google.golang.org/protobuf/proto"
+)
+
+func (e *Engine) NewExecOutput(blockType string, block *bstream.Block, clock *pbsubstreams.Clock, cursor *bstream.Cursor) (execout.ExecOutput, error) {
+	blkBytes, err := block.Payload.Get()
+	if err != nil {
+		return nil, fmt.Errorf("getting block %d %q: %w", block.Number, block.Id, err)
+	}
+	clockBytes, err := proto.Marshal(clock)
+	if err != nil {
+		return nil, fmt.Errorf("getting block %d %q: %w", block.Number, block.Id, err)
+	}
+	return &ExecOutputCache{
+		engine: e,
+		clock:  clock,
+		cursor: cursor,
+		values: map[string][]byte{
+			blockType:                blkBytes,
+			"sf.substreams.v1.Clock": clockBytes,
+		},
+	}, nil
+}
```

Comment:
This whole function's code is duplicated by the two implementations. It can be factored into a simple function, and called by the caller before calling NewExecOutput.